nri_device_injector/nri_device_injector_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2023 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseDevices(t *testing.T) {
```

Comment:
Not a blocker, but something else we could do: set up some test device nodes with known attributes, and validate `device.toNRIDevice` as well.

Reply:
By `set up some test device nodes with known attributes`, do you mean manual test on GKE nodes or some tests like https://github.com/opencontainers/runc/blob/main/libcontainer/devices/device_unix_test.go?

Reply:
You can create device node files (at any path) using mknod(2) or [`unix.Mknod`](https://pkg.go.dev/golang.org/x/sys/unix#Mknod).  Then you should be able to read them with `unix.Lstat` inside `device.toNRIDevice`.

(mknod(2) requires `CAP_MKNOD`, so that may mean the easiest way to run the test is with `sudo`.  My usual preferred pattern for something like this is to build a test binary using `go test -c` and then run the binary with `sudo`, rather than `sudo go test` that also runs the Go toolchain with privilege.  There's an example of that [here](https://github.com/samuelkarp/runj/blob/4c1dfb645be03f0deb1faae5a551ce6039368d44/Makefile#L31-L39).  It can also be useful to specifically mark a test as requiring root with something like [containerd's `testutil.RequiresRoot`](https://github.com/containerd/containerd/blob/ced9b18c231a28990617bc0a4b8ce2e81ee2ffa1/pkg/testutil/helpers_unix.go#L29-L36) and a [`-test.root` command line flag](https://github.com/containerd/containerd/blob/ced9b18c231a28990617bc0a4b8ce2e81ee2ffa1/Makefile#L214).)

Probably best to do this all in a follow-up PR so you don't block submission on this.

Reply:
I see, thanks Sam, I'll create another PR for this.